server/models/GameLogic.js

```diff
@@ -171,86 +169,72 @@ class GameLogic {
+        } else if (this.playerHasPlayedCard(username)) {
+            throw Error("You cannot play more than one card");
+        } else {
             const card = { username, cardId };
             this.playedCards.push(card);
             socket.emitPlayedCards(this.roomId, this.getHiddenPlayedCards());
+            socket.emitPlayedCard(username, card);
             this.getCardsByUsername(username).find(playedCard => playedCard.cardId === cardId).played = true;
             this.markTurnAsFinished(username);
             if (this.allPlayersPlayedCard()) {
-                this.clearTimeouts();
+                clearTimeout(this.playCardTimeout);
                 this.clearFinishedTurn();
-                this.setStatus(statusTypes.WAITING_FOR_VOTES);
-                socket.emitPlayedCards(this.roomId, this.getPlayedCards());
-                socket.emitStatus(this.roomId, this.status);
-                socket.promptPlayersVote(this.roomId, this.currentPlayer, promptDuration);
-                this.voteTimeout = setTimeout(() => this.emitVotes(), promptDuration);
+                this.emitPlayedCards();
             }
-        } else {
-            // Cannot play card, the player has already played a card, or the game status is not
-            // appropriate for the request, server is responsible for generating an error.
-            throw Error("You cannot play more than one card, or now is not the right time to play a card.");
         }
     }
 
+    /* Emit the played cards for voting */
+    emitPlayedCards() {
+        this.setStatus(statusTypes.WAITING_FOR_VOTES);
+        socket.emitPlayedCards(this.roomId, this.getPlayedCards());
+        socket.emitStatus(this.roomId, this.status);
+        socket.promptPlayersVote(this.roomId, this.currentPlayer, promptDuration);
+        this.voteTimeout = setTimeout(this.emitVotes.bind(this), promptDuration);
+    };
+
     /* Vote for a card */
     voteCard(username, cardId) {
-        if (this.status === statusTypes.WAITING_FOR_VOTES && !this.playerHasVoted(username)) {
+        if (this.status !== statusTypes.WAITING_FOR_VOTES) {
+            throw Error("Now is not the right time to vote");
+        } else if (this.playerHasVoted(username)) {
+            throw Error("You cannot vote for a card more than once");
+        } else {
             const vote = { username, cardId };
             this.votes.push(vote);
             this.markTurnAsFinished(username);
             if (this.allPlayersVoted()) {
-                this.clearTimeouts();
-                this.setStatus(statusTypes.DISPLAY_ALL_VOTES);
-                socket.emitStatus(this.roomId, this.status);
-                socket.emitAllVotes(this.roomId, this.votes);
-                this.calcScores();
-                this.newCard();
-                this.nextRoundTimeout = setTimeout(() => this.nextRound(), nextRoundDuration);
+                clearTimeout(this.voteTimeout);
```

Comment:
this.clearTimeout?

Reply:
`this.clearTimeouts` is our function, `clearTimeout` is the js thing for clearing a timeout